certloader/spiffe_tls_config_test.go

```diff
@@ -0,0 +1,107 @@
+package certloader
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/spiffe/go-spiffe/spiffetest"
+	"github.com/stretchr/testify/require"
+)
+
+type testLogger bytes.Buffer
+
+func (l *testLogger) Printf(format string, args ...interface{}) {
+	fmt.Fprintf((*bytes.Buffer)(l), format, args...)
```

Comment:
Nit: Could map this to `t.Logf` from the `testing` package maybe, would produce slightly cleaner output I think.

Reply:
oops, i idea here was to assert that certain things were logged but I forgot to implement the checks.